thrifty-schema/src/main/java/com/microsoft/thrifty/schema/UserElement.java

```diff
@@ -0,0 +1,40 @@
+package com.microsoft.thrifty.schema;
+
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * Represents data common to user-defined elements of a Thrift program.
+ */
+public interface UserElement {
+    /**
+     * Gets the name of the element.
+     */
+    String name();
+
+    /**
+     * Gets the {@link Location} where the element is defined.
+     */
+    Location location();
+
+    /**
+     * Gets the documentation comments of the element, or an empty string.
+     */
+    String documentation();
+
+    /**
+     * Gets an immutable map containing any annotations present on the element.
+     */
+    ImmutableMap<String, String> annotations();
+
+    /**
+     * Gets a value indicating whether the element contains non-empty Javadoc.
+     */
+    boolean hasJavadoc();
```

Comment:
The particulars are lost to the mists of time; I think it was safer/more convenient to assume that `javadoc()` would never return null.